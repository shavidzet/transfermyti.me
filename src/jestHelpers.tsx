/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
