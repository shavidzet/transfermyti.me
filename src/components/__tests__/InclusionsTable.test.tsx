import React from "react";
import { render } from "~/jestHelpers";
import InclusionsTable from "../InclusionsTable";

describe("the <InclusionsTable> component", () => {
  test("matches its snapshot with valid props", () => {
    const wrapper = render(<InclusionsTable />);

    expect(wrapper.container.firstElementChild).toMatchSnapshot();
  });
});
