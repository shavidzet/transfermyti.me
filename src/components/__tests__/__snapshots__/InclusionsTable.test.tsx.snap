// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <InclusionsTable> component matches its snapshot with valid props 1`] = `
.emotion-0 {
  border-spacing: 0;
  table-layout: fixed;
  width: 100%;
  border: 1px solid rgb(30,120,161);
}

.emotion-0 td,
.emotion-0 th {
  padding: 0.5rem;
}

.emotion-0 thead th,
.emotion-0 td {
  text-align: left;
}

.emotion-0 thead tr:first-of-type th {
  border-top: none;
}

.emotion-0 [data-include='true'] {
  border-right: none;
  text-align: center;
  width: 6rem;
}

.emotion-0 tfoot td:last-of-type {
  border-right: none;
}

.emotion-0 tbody tr:first-of-type td {
  border-top: none;
}

.emotion-0 th,
.emotion-0 tfoot td {
  background: rgb(30,120,161);
  border-right: 1px solid rgb(255,255,255);
  color: rgb(255,255,255);
  font-weight: 600;
}

.emotion-0 thead th {
  border-bottom: 1px solid rgb(255,255,255);
}

.emotion-0 tbody td {
  background: rgb(255,255,255);
  border-right: 1px solid rgb(30,120,161);
  border-top: 1px solid rgb(30,120,161);
}

.emotion-0 tbody tr:last-of-type td {
  border-bottom: 1px solid rgb(255,255,255);
}

@media (max-width:32rem) {
  .emotion-0 td,
  .emotion-0 th {
    font-size: 0.875rem;
    padding: 0.375rem;
  }

  .emotion-0 [data-include='true'] {
    width: 4.25rem;
  }
}

<table
  class="emotion-0"
  role="grid"
/>
`;
